frontend/views/components/ContributionItem.vue

```diff
@@ -0,0 +1,155 @@
+<template lang='pug'>
+.c-contribution-item
+  i(:class='iconClass')
+
+  template(v-if='hasWhoElse')
+    transition(:name='showEdit')
+      div(
+        v-if='isVisible'
+        key='visible'
+      )
+        .c-contribution-list(v-html='listOfName')
+
+        i18n.is-unstyled.is-link-inherit.link(
+          tag='button'
+          type='button'
+          @click='isVisible = !isVisible'
+        ) Hide
+
+      div(
+        v-else
+        key='hidden'
+        @click='isVisible = !isVisible'
+        v-html='contribution'
+      )
+
+  .c-contribution-list(
+    v-else=''
+    v-html='contribution'
+  )
+</template>
+
+<script>
+import L from '@view-utils/translations.js'
+
+export default {
+  name: 'ContributionItem',
+  props: {
+    who: [String, Array],
+    what: String,
+    action: {
+      type: String,
+      validator: function (value) {
+        // The value must match one of these strings
+        return ['RECEIVING', 'GIVING'].indexOf(value) !== -1 || null
+      },
+      default: 'RECEIVING'
+    },
+    type: {
+      type: String,
+      validator: function (value) {
+        // The value must match one of these strings
+        return ['NON_MONETARY', 'MONETARY'].indexOf(value) !== -1
+      }
```

Comment:
Is there a reason for the inconsistency between the validations for the properties? I.e. `action` has a default value while `type` does not

Reply:
fixed